### ISSUE TITLE
[Jira] "Agile" methods don't use the “api_root” instance attribute, as most other methods do

### ISSUE THREAD

**Issue by super-vanilla-bear** [atlassian-api/atlassian-python-api#1423]
While building an application that implemented [Atlassian OAuth2 (3LO)](https://developer.atlassian.com/cloud/jira/platform/oauth-2-3lo-apps/) authorization, I encountered a situation where the two methods I used for [Jira](https://github.com/atlassian-api/atlassian-python-api/blob/master/atlassian/jira.py) entities behaved differently - one worked, and the other returned a 404 error. These methods are `Jira().jql` and `Jira().get_all_sprints_from_board` correspondingly.

A feature of initializing the Jira client with a token obtained via the OAuth2 flow is that the addresses that should be used to make API calls differ from the "standard" one, so [as described in this document](https://developer.atlassian.com/cloud/jira/platform/oauth-2-3lo-apps/#3-2-construct-the-request-url):
instead of `https://your-domain.atlassian.net/rest/api/2/{api_endpoint}`
you should use `https://api.atlassian.com/ex/jira/{cloudid}/rest/api/2/{api_endpoint}`

The Jira client may be initialized with OAuth2 creds, and moreover, it accommodates the address feature mentioned above, namely, by passing `api_root=f'/ex/jira/{cloudid}/rest/api'` instead of using the default value of [`api_root="rest/api"`](https://github.com/atlassian-api/atlassian-python-api/blob/42ed0b022ec5d13416d79f4118e655f536247d07/atlassian/rest_client.py#L54) on its init.

And this allows you to successfully send requests using the `Jira().jql` method, since it calls the [`Jira().resource_url`](https://github.com/atlassian-api/atlassian-python-api/blob/master/atlassian/rest_client.py#L226-L231) method inside, which takes into account the value of `Jira().api_root`

At the same time, the `Jira().get_all_sprints_from_board` method, like all the methods next to it, located in the informal "Agile (Formerly Greenhopper) REST API" section, [uses hardcoded paths](https://github.com/atlassian-api/atlassian-python-api/blob/master/atlassian/jira.py#L5056-L5057) that ignore the possible need to have a non-standard path prefix.

So, to summarize, the standard behavior of the `Jira().get_all_sprints_from_board` method generates the following URL path:
```
'rest/agile/1.0/board/{board_id}/sprint'
```
Whereas (to be functional in the described circumstances) it should be the following:
```
'ex/jira/{cloudid}/rest/agile/1.0/board/{board_id}/sprint'
```

The situation is similar to https://github.com/atlassian-api/atlassian-python-api/issues/744 , and for another method section in the Jira class, namely "Tempo Account REST API".

This ticket is to some extent a question, is this the expected behavior and can I miss something, or is this not the expected behavior?
If the latter, I could create a PR with a fix for this

**Comment by super-vanilla-bear** [comment]
So, as I said, I can take it and make a PR, if it's really an issue and not my misunderstanding of the structure/concept :)

**Comment by gonchik** [comment]
@super-vanilla-bear  Thank you  very much
